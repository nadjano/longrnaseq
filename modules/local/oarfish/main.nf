process OARFISH {
    tag "$meta.id"
    label 'process_single'

    cache 'lenient'

    // TODO nf-core: List required Conda package(s).
    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'community.wave.seqera.io/library/oarfish:0.9.0--4958b2b67f1dd777':
        'biocontainers/oarfish:0.8.1--h5ca1c30_0' }"


    input:
    // TODO nf-core: Where applicable all sample-specific information e.g. "id", "single_end", "read_group"
    //               MUST be provided as an input via a Groovy Map called "meta".
    //               This information may not be required in some instances e.g. indexing reference genome files:
    //               https://github.com/nf-core/modules/blob/master/modules/nf-core/bwa/index/main.nf
    // TODO nf-core: Where applicable please provide/convert compressed files as input/output
    //               e.g. "*.fastq.gz" and NOT "*.fastq", "*.bam" and NOT "*.sam" etc.
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("${prefix}")        , emit: results
    tuple val(meta), path("${meta.id}.ambig_info.tsv")        , emit: ambig_info
    tuple val(meta), path("*.meta_info.json")  , emit: json_info
    tuple val(meta), path(".command.log")            , emit: log
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir $prefix && oarfish $args \\
            -j $task.cpus \\
            --alignments $bam \\
            -o $prefix \\
            --filter-group no-filters \\
            --write-assignment-probs \\
            --score-threshold 1.0

    # Add the transcript ids to the ambig_info.tsv file
    paste <(awk '{print \$1}' "${meta.id}.quant") \\
      <(awk '{print \$1 "\\t" \$2}' "${meta.id}.ambig_info.tsv") \\
      > "${meta.id}.ambig_info.tmp" && \\
    mv "${meta.id}.ambig_info.tmp" "${meta.id}.ambig_info.tsv"

    # Rename .qant file to qant.sf to match salmon output
    mv ${prefix}.quant ${prefix}/quant.sf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        oarfish: \$(oarfish --version |& sed '1!d ; s/oarfish //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    // TODO nf-core: A stub section should mimic the execution of the original module as best as possible
    //               Have a look at the following examples:
    //               Simple example: https://github.com/nf-core/modules/blob/818474a292b4860ae8ff88e149fbcda68814114d/modules/nf-core/bcftools/annotate/main.nf#L47-L63
    //               Complex example: https://github.com/nf-core/modules/blob/818474a292b4860ae8ff88e149fbcda68814114d/modules/nf-core/bedtools/split/main.nf#L38-L54
    """
    mkdir -p $prefix
    touch ${prefix}.log
    touch ${prefix}.meta_info.json
    touch ${prefix}.ambig_info.json


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        oarfish: \$(oarfish --version |& sed '1!d ; s/oarfish //')
    END_VERSIONS
    """
}
