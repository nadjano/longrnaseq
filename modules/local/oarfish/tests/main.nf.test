// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test oarfish
nextflow_process {

    name "Test Process OARFISH"
    script "../main.nf"
    process "OARFISH"

    tag "modules"
    tag "modules_nfcore"
    tag "oarfish"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    // test("sarscov2 - bam") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

    //     when {
    //         process {
    //             """
    //             // TODO nf-core: define inputs of the process here. Example:

    //             input[0] = [
    //                 [ id:'test'], // meta map
    //                 file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
    //             ]
    //             """
    //         }
    //     }

    //     then {
    //         assertAll(
    //             { assert process.success },
    //             { assert path(process.out.json_info.get(0).get(1)).exists() },
    //             { assert path(process.out.results.get(0).get(1)).exists() },
    //             { assert snapshot(
    //                 process.out.versions,
    //                 process.out.lib_format_counts
    //             ).match() }
    //         )
    //     }

    // }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.json_info.get(0).get(1)).exists() },
                { assert path(process.out.results.get(0).get(1)).exists() },
                { assert snapshot(
                    process.out.versions,
                    process.out.lib_format_counts
                ).match() }
            )
        }


    }

}
