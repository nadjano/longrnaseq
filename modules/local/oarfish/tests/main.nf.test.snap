{
    "sarscov2 - bam - stub": {
        "content": [
            [
                
            ],
            null
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-29T13:16:46.808467544"
    },
    "sarscov2 - bam": {
        "content": [
            [
                
            ],
            null
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-29T13:16:39.802533999"
    }
}